Algorithm.CSharp/CustomDataUsingMapFileRegressionAlgorithm.cs

```diff
@@ -0,0 +1,146 @@
+            SetCash(100000);
+
+            // AOL renames to TWX in 2003
+            _symbol = AddData<SECReport8K>(Ticker, Resolution.Daily).Symbol;
+            AddEquity(Ticker, Resolution.Daily);
+        }
+
+        /// <summary>
+        /// Checks to see if the stock has been renamed, and places an order once the symbol has changed
+        /// </summary>
+        /// <param name="slice"></param>
+        public override void OnData(Slice slice)
+        {
+            if (slice.SymbolChangedEvents.ContainsKey(_symbol))
+            {
+                // Check to see if it was renamed on the 16th
+                _changedSymbol = Time.Date == new DateTime(2003, 10, 16);
+                Log($"{Time} - Ticker changed from: {slice.SymbolChangedEvents[_symbol].OldSymbol} to {slice.SymbolChangedEvents[_symbol].NewSymbol}");
+            }
+
+            foreach (var report in slice.Get<SECReport8K>())
+            {
+                if (!_properSymbolBeforeRename)
+                {
+                    _properSymbolBeforeRename = report.Key.Value == "AOL" && Time < new DateTime(2003, 10, 16);
+                }
+                if (!_properSymbolAfterRename)
+                {
+                    _properSymbolAfterRename = report.Key.Value == "TWX" && Time >= new DateTime(2003, 10, 16);
+                }
+
+                Log($"{Time} - Received 8-K report for {report.Key.Value}");
+            }
+        }
+        
+        /// <summary>
+        /// Final step of the algorithm
+        /// </summary>
+        public override void OnEndOfAlgorithm()
+        {
+            if (!_changedSymbol)
+            {
+                throw new Exception("The ticker did not rename throughout the course of its life even though it should have");
+            }
+            if (!_properSymbolBeforeRename)
+            {
+                throw new Exception("The SEC report data never renamed to its old ticker");
+            }
+            if (!_properSymbolAfterRename)
+            {
+                throw new Exception("The SEC report data never renamed back to its present-day ticker");
+            }
+        }
+
+        /// <summary>
+        /// This is used by the regression test system to indicate if the open source Lean repository has the required data to run this algorithm.
+        /// </summary>
+        public bool CanRunLocally { get; } = false;
+
+        /// <summary>
+        /// This is used by the regression test system to indicate which languages this algorithm is written in.
+        /// </summary>
+        public Language[] Languages { get; } = { Language.CSharp, Language.Python };
+
+        /// <summary>
+        /// This is used by the regression test system to indicate what the expected statistics are from running the algorithm
+        /// </summary>
+        public Dictionary<string, string> ExpectedStatistics => new Dictionary<string, string>
+        {
+            {"Total Trades", "0"},
```

Comment:
In general, I think it's best to create at least one trade... I forget _why_ this was the previous guidance, but I do remember it being the recommendation at one point in time.